src/index.js

```diff
@@ -22,23 +22,33 @@ export default function chunkedRequest(options) {
 
   let prevChunkSuffix = "";
 
-  function processRawChunk(rawChunk) {
+  function processRawChunk(rawChunk, isFinalChunk = false) {
     let parsedChunks = null;
     let parseError = null;
     let suffix = "";
 
     try {
-      [ parsedChunks, suffix ] = chunkParser(rawChunk, prevChunkSuffix);
+      [ parsedChunks, suffix ] = chunkParser(rawChunk, prevChunkSuffix, isFinalChunk);
       prevChunkSuffix = suffix || "";
     } catch (e) {
       parseError = e;
       parseError.rawChunk = rawChunk;
       parseError.prevChunkSuffix = prevChunkSuffix;
     } finally {
-      onChunk(parseError, parsedChunks);
+      if (parseError || (parsedChunks!=null && parsedChunks.length > 0) ) {
+        onChunk(parseError, parsedChunks);
+      }
     }
   }
 
+  function processRawComplete(rawComplete) {
+    if( prevChunkSuffix != "") {
```

Comment:
nit: formatting (exactly one space after `if` keyword and no space after opening `(`)